asistente_ladm_col/utils/qgis_utils.py

```diff
@@ -1041,24 +1046,118 @@ def show_etl_model(self, db, input_layer, ladm_col_layer_name, geometry_type=Non
         if model:
             automatic_fields_definition = self.check_if_and_disable_automatic_fields(db, ladm_col_layer_name)
 
-            mapping = get_refactor_fields_mapping(ladm_col_layer_name, self)
+            if save_field_mapping is not None:
+                if self.func_load_field_mapping(ladm_col_layer_name, save_field_mapping) is not None:
+                    mapping = self.func_load_field_mapping(ladm_col_layer_name, save_field_mapping)
+                else:
+                    self.message_emitted.emit(
+                        QCoreApplication.translate("QGISUtils",
+                                                   "Field mapping was not found and cannot be charged!"),
+                        Qgis.Info)
+                    mapping = get_refactor_fields_mapping(ladm_col_layer_name, self)
+            else:
+                mapping = get_refactor_fields_mapping(ladm_col_layer_name, self)
+
             self.activate_layer_requested.emit(input_layer)
             params = {
                 'INPUT': input_layer.name(),
                 'mapping': mapping,
                 'output': output.name()
             }
+
+            start_feature_count = output.featureCount()
             processing.execAlgorithmDialog("model:ETL-model", params)
+            finish_feature_count = output.featureCount()
 
             self.check_if_and_enable_automatic_fields(db,
                                                       automatic_fields_definition,
                                                       ladm_col_layer_name)
+
+            return start_feature_count, finish_feature_count
         else:
             self.message_emitted.emit(
                 QCoreApplication.translate("QGISUtils",
                                            "Model ETL-model was not found and cannot be opened!"),
                 Qgis.Info)
 
+    def func_load_field_mapping(self,ladm_col_layer_name, save_field_mapping):
+        path_file_field_mapping = os.path.join(FIELD_MAPPING_PATH,'{}.{}'.format(save_field_mapping,
+                                                "txt"))
+
+        field_mapping_list = []
+
+        with open(path_file_field_mapping) as file_field_mapping:
+            try:
+                mapping = ast.literal_eval(file_field_mapping.read())
+            except:
+                mapping = None
+
+        return mapping
+
+    def func_save_field_mapping(self,ladm_col_layer_name):
+        files_mapping_path = os.path.join(os.path.expanduser('~'), 'Asistente-LADM_COL', 'field_mapping')
```

Comment:
Then, what is `FIELD_MAPPING_PATH` for?

Reply:
yes, we have to use FIELD_MAPPING_PATH, i forget it